google/cloud/storage/tests/bucket_integration_test.cc

```diff
@@ -500,6 +501,105 @@ TEST_F(BucketIntegrationTest, NotificationsCRUD) {
   client.DeleteBucket(bucket_name);
 }
 
+TEST_F(BucketIntegrationTest, NoexNotificationsCRUD) {
+  auto project_id = BucketTestEnvironment::project_id();
+  std::string bucket_name = MakeRandomBucketName();
+  noex::Client client;
+
+  // TODO(#1681) - just use the noex::Client when it implements bucket CRUD
+  //     operations (both #1681 and #1679 are needed for this).
+  Client ex_client;
+
+  // Create a new bucket to run the test.
+  auto meta = ex_client.CreateBucketForProject(bucket_name, project_id,
+                                               BucketMetadata());
+
+  auto current_notifications = ex_client.ListNotifications(bucket_name);
+  EXPECT_TRUE(current_notifications.empty())
+      << "Test aborted. Non-empty notification list returned from newly"
+      << " created bucket <" << bucket_name
+      << ">. This is unexpected because the bucket name is chosen at random.";
+
+  StatusOr<NotificationMetadata> create_err = client.CreateNotification(
+      bucket_name, BucketTestEnvironment::topic(), payload_format::JsonApiV1(),
+      NotificationMetadata().append_event_type(event_type::ObjectFinalize()));
+  ASSERT_TRUE(create_err.ok())
+      << "Unexpected CreateNotification() failure in integration test: "
+      << create_err.status();
+
+  NotificationMetadata create = std::move(create_err).value();
```

Comment:
While the `std::move(sor).value()` pattern is "right", it's a little verbose and I wonder if there's maybe a simpler patter that we might use and encourage in this code. In particular, I'm thinking of functions that return a StatusOr<T>, and then use it like a smart pointer via op*(), op->() and op bool(). So this chunk of code might look something like:

```cc
StatusOr<NotificationMetadata> create = client.CreateNotification(...);  // Or even use 'auto' here
ASSERT_TRUE(create) << "Error: " << create.status();
EXPECT_EQ(payload_format::JsonApiV1(), create->payload_format());
EXPECT_THAT(create->topic(), HasSubstr(...));
```
This is a common pattern that's used and encouraged in some other StatusOr code that I've seen. I tend to like it because it's efficient (no moves or copies), simple, and looks like normal smart pointer usage that people are familiar with.

Thoughts?

Reply:
I changed the implementation so we can take a look.  I fear it looks a bit too magical, but opinions may vary.